test/integration_test.py

```diff
@@ -2053,17 +2057,30 @@ def test_response_security_headers(self):
                     else:
                         response = requests.get(str(endpoint / path))
                         response.raise_for_status()
-                        expected = AzulChaliceApp.security_headers | expected_headers
-                        # FIXME: Add a CSP header with a nonce value to text/html responses
-                        #        https://github.com/DataBiosphere/azul-private/issues/6
+                        expected = AzulChaliceApp.security_headers() | expected_headers
+                        # These paths have a CSP that differs from the default
+                        # by including randomly generated 'nonce' values.
                         if path in ['/', '/oauth2_redirect']:
-                            del expected['Content-Security-Policy']
+                            nonce = set()
+                            for directive in response.headers['Content-Security-Policy'].split(';'):
+                                match = directive_re.fullmatch(directive)
+                                self.assertIsNotNone(match)
+                                name, values = match.groups()
```

Comment:
Why do you use plural for `values`?

Reply:
This was accidentally left in from a prior approach where I was splitting the value on spaces.